src/ajax.js

```diff
@@ -270,7 +271,22 @@ function ajaxConvert( s, response, jqXHR, isSuccess ) {
 		}
 	}
 
-	return { state: "success", data: response };
+	if (response && response.then) {
```

Comment:
Yep, completely aware, and already on my checklist of things to do above. :) In making this PR I intend to close the gaps in test coverage that allowed my async converter to be present and still have all tests pass. I appreciate the detailed review!